submission/submission-webapp/src/main/kotlin/ac/uk/ebi/biostd/submission/domain/helpers/FireFilesSource.kt

```diff
@@ -73,5 +77,12 @@ fun FireApiFile.asFireBioFile(path: String, attributes: List<Attribute>): FireFi
         fireId = fireOid,
         md5 = objectMd5,
         size = objectSize.toLong(),
+        type = fileType,
         attributes = attributes.toExtAttributes(FILES_RESERVED_ATTRS)
     )
+
+private val FireApiFile.fileType: ExtFileType
+    get(): ExtFileType {
+        val type = metadata!!.first { it.key == FIRE_BIO_FILE_TYPE }.value
+        return if (type == FILE.value) FILE else DIR
```

Comment:
I think the default need to be file and not directory.

Reply:
Nice catch, even better I implemented it from the enum itself. The compiler is our friend